ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/CreateTableEvent.java

```diff
@@ -56,16 +57,28 @@ public HiveMetaStoreAuthzInfo getAuthzContext() {
     return ret;
   }
 
-  private List<HivePrivilegeObject> getInputHObjs() { return Collections.emptyList(); }
+  private List<HivePrivilegeObject> getInputHObjs() {
+    List<HivePrivilegeObject> ret   = new ArrayList<>();
+    PreCreateTableEvent       event = (PreCreateTableEvent) preEventContext;
+    Table                     table = event.getTable();
+    String                    uri   = getSdLocation(table.getSd());
+
```

Comment:
Shouldn't we also add the table itself to this set? Looks like we are adding just the location check